Content.Server/Nyanotrasen/RoundNotifications/RoundNotificationsSystem.cs

```diff
@@ -0,0 +1,178 @@
+    [Dependency] private readonly IGameMapManager _gameMapManager = default!;
+
+    private ISawmill _sawmill = default!;
+    private readonly HttpClient _httpClient = new();
+
+    private string _webhookUrl = String.Empty;
+    private string _roleId = String.Empty;
+    private bool _roundStartOnly;
+    private string _serverName = string.Empty;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestart);
+        SubscribeLocalEvent<RoundStartedEvent>(OnRoundStarted);
+        SubscribeLocalEvent<RoundEndedEvent>(OnRoundEnded);
+
+        _config.OnValueChanged(CCVars.DiscordRoundWebhook, value => _webhookUrl = value, true);
+        _config.OnValueChanged(CCVars.DiscordRoundRoleId, value => _roleId = value, true);
+        _config.OnValueChanged(CCVars.DiscordRoundStartOnly, value => _roundStartOnly = value, true);
+        _config.OnValueChanged(CVars.GameHostName, OnServerNameChanged, true);
+
+        _sawmill = IoCManager.Resolve<ILogManager>().GetSawmill("notifications");
+    }
+
+    private void OnServerNameChanged(string obj)
+    {
+        _serverName = obj;
+    }
+
+    private void OnRoundRestart(RoundRestartCleanupEvent e)
+    {
+        if (String.IsNullOrEmpty(_webhookUrl))
+            return;
+
+        var text = Loc.GetString("discord-round-new");
+
+        SendDiscordMessage(text);
+    }
+
+    private void OnRoundStarted(RoundStartedEvent e)
+    {
+        if (String.IsNullOrEmpty(_webhookUrl))
+            return;
+
+        var map = _gameMapManager.GetSelectedMap();
+        var mapName = map?.MapName ?? Loc.GetString("discord-round-unknown-map");
+        var text = Loc.GetString("discord-round-start",
+            ("id", e.RoundId),
+            ("map", mapName));
+
+        SendDiscordMessage(text);
+    }
+
+    private void OnRoundEnded(RoundEndedEvent e)
+    {
+        if (String.IsNullOrEmpty(_webhookUrl) || _roundStartOnly)
+            return;
+
+        var text = Loc.GetString("discord-round-end",
+            ("id", e.RoundId),
+            ("hours", Math.Truncate(e.RoundDuration.TotalHours)),
+            ("minutes", e.RoundDuration.Minutes),
+            ("seconds", e.RoundDuration.Seconds));
+
+        SendDiscordMessage(text);
+    }
+
+    private async void SendDiscordMessage(string text)
+    {
+
+        // Limit server name to 1500 characters, in case someone tries to be a little funny
+        var serverName = _serverName[..Math.Min(_serverName.Length, 1500)];
+        var message = "";
+        if (!String.IsNullOrEmpty(_roleId))
+            message = $"<@&{_roleId}>";
+
+        // Build the embed
+        var payload = new WebhookPayload
+        {
+            Message = message,
+            Embeds = new List<Embed>
+            {
+                new()
+                {
+                    Description = text,
+                    Color = 0x41F097,
```

Comment:
Hardcoded :/

Reply:
You talking about the color or the title?

Reply:
color

Reply:
not a big deal really, if you wanna push out a fix real fast go ahead, otherwise im just gonna merge it anyway